pkg/provider/akeyless/akeyless_api.go

```diff
@@ -240,6 +248,117 @@ func (a *akeylessBase) getCloudID(provider, accTypeParam string) (string, error)
 	return cloudID, err
 }
 
+func (a *akeylessBase) getK8SServiceAccountJWT(ctx context.Context, kubernetesAuth *esv1beta1.AkeylessKubernetesAuth) (string, error) {
+	if kubernetesAuth.ServiceAccountRef != nil {
+		// Kubernetes <v1.24 fetch token via ServiceAccount.Secrets[]
+		jwt, err := a.getJWTFromServiceAccount(ctx, kubernetesAuth.ServiceAccountRef)
```

Comment:
To make that simpler, this `getJWTFromServiceAccount()` (pulling it from secrets) can be omitted completely unless `Akeyless` cares about the `iss` claim. I've removed that approach from the Kubernetes provider as well. AFAIK only vault is (or better was <1.9) sensible about the `iss` claim.

Reply:
for now, Akeyless performs ISS Claim validation, so for now I prefer to leave it as is